File: TodoList/TodoList.Server/Models/TodoItem.cs
using System.ComponentModel.DataAnnotations;

namespace TodoList.Server.Models
{
    public class TodoItem
    {
        public int Id { get; set; }

        [Required]
        public bool IsComplete { get; set; }

        [Required]
        public string Name { get; set; }

        public string? DateAndTime { get; set; }
        public string? Secret { get; set; }
    }
}

File: TodoList/todolist.client/src/app/models/todoItem.ts
export class TodoItem {
  id: number;
  isComplete: boolean;
  name: string;
  dateAndTime?: string | null;
  constructor(id: number, isComplete: boolean, name: string, dateAndTime?: string) {
    this.id = id;
    this.isComplete = isComplete;
    this.name = name;
    this.dateAndTime = dateAndTime;
  }
}
